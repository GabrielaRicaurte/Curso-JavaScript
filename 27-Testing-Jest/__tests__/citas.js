import Citas from '../js/classes/Citas';


describe('Probar la clase de citas', ()=> {
    
    const citas = new Citas();

    const id = Date.now();

    test('Agregar nueva cita', ()=>{
        const citaObj = {
            id,
            mascota: 'Franchisco',
            propietario: 'Gaby',
            telefono: '04164027739',
            fecha: '10/12',
            hora:'02:26',
            sintomas: 'solo duerme'
        }

        citas.agregarCita(citaObj);

        // Prueba

        expect(citas).toMatchSnapshot();
        
    });

    test('actualizar cita', () => { 
        const citaActualizada = {
            id,
            mascota: 'Nuevo Nombre',
            propietario: 'Gaby',
            telefono: '04164027739',
            fecha: '10/12',
            hora:'02:26',
            sintomas: 'solo duerme'
        }

        citas.editarCita(citaActualizada);
        
        expect(citas).toMatchSnapshot();
    });

    test('Eliminar Cita', ()=> {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    })
});