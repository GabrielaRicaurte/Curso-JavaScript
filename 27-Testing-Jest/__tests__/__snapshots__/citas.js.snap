// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Agregar nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10/12",
      "hora": "02:26",
      "id": 1697742743003,
      "mascota": "Franchisco",
      "propietario": "Gaby",
      "sintomas": "solo duerme",
      "telefono": "04164027739",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar Cita 1`] = `
Citas {
  "citas": [],
}
`;

exports[`Probar la clase de citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10/12",
      "hora": "02:26",
      "id": 1697742743003,
      "mascota": "Nuevo Nombre",
      "propietario": "Gaby",
      "sintomas": "solo duerme",
      "telefono": "04164027739",
    },
  ],
}
`;
